modules/core/src/v2/wallets.ts

```diff
@@ -336,10 +341,26 @@ export class Wallets {
       }
 
       if (!_.isUndefined(params.gasPrice)) {
-        if (!_.isNumber(params.gasPrice)) {
+        const gasPriceBN = new BigNumber(params.gasPrice);
+        if (gasPriceBN.isNaN()) {
           throw new Error('invalid gas price argument, expecting number');
         }
-        walletParams.gasPrice = params.gasPrice;
+        walletParams.gasPrice = gasPriceBN.toString();
+      }
+
+      if (!_.isUndefined(params.eip1559) && !_.isEmpty(params.eip1559)) {
```

Comment:
if gasPrice and eip1559 are passed what happens?

Reply:
Adding validation in case of sending both fees